app/graphql/record_loader.rb

```diff
@@ -4,7 +4,21 @@ def initialize(model)
   end
 
   def perform(ids)
-    @model.where(id: ids).each { |record| fulfill(record.id, record) }
+    authorized_scope.where(id: ids).each { |record| fulfill(record.id, record) }
     ids.each { |id| fulfill(id, nil) unless fulfilled?(id) }
   end
+
+  def load_by_global_id(global_id)
+    id = Foreman::GlobalId.decode(global_id).last.to_i
+    load(id)
+  end
+
+  private
+
+  def authorized_scope
+    return @model unless @model.respond_to?(:authorized)
+
+    permission_name = @model.find_permission_name(:view)
+    @model.authorized_as(User.current, permission_name)
```

Comment:
Should this use the context current_user? (I'm not really sure I understand why in certain places we pass the context around and others not)

Reply:
GraphQL expected that the current user is passed via the context, but we rely on `User.current` for writing Audits etc. So we basically need both.
In the future, we should probably unify this, but I think right now we can live with providing both options.